File: cypher.py
#!/usr/bin/python3

import os, sys

def c_cypher(n, overwrite):
    n %= 26
    if os.path.isfile(f"zasifrovana({n}).txt") and not overwrite:
        print("File already exists. If you want to overwrite, run the command with --force-overwrite or -f")
        sys.exit(1)

    lower_case = range(ord('a'), ord('z')+1)
    upper_case = range(ord('A'), ord('Z')+1)
    try:
        with open(f"zasifrovana({n}).txt", "w") as out_file:
            for line in sys.stdin:
                result = ""
                for char in line:
                    if ord(char) in lower_case:
                        result += chr((ord(char) + n - ord('a')) % 26 + ord('a'))
                    elif ord(char) in upper_case:
                        result += chr((ord(char) + n - ord('A')) % 26 + ord('A'))
                    else:
                        result += char
                print(result, end='')
                out_file.write(result)
    except:
        print("Error has occured.")
    finally:
        out_file.close()

def print_help():
    print("Usage:")
    print("a) python3 cypher.py YOUR_INT [OPTION] < YOUR_FILE_TO_ENCRYPT")
    print("b) python3 cypher.py YOUR_SOURCE_CHAR YOUR_DESTINATION_CHAR [OPTION] < YOUR_FILE_TO_ENCRYPT\n")
    print("[OPTION]:")
    print("-h --help")
    print("   prints usage (this).\n")
    print("-f --force-overwrite")
    print("   if destination file exists, overwrites it.")
    sys.exit(2)

def recommend_help():
    print("Bad arguments.")
    print("Type: \"python3 cypher.py -h\" for help.")
    sys.exit(2)

def in_alphabet(c):
    return (ord(c) >= ord('a') and ord(c) <= ord('z')) or (ord(c) >= ord('A') and ord(c) <= ord('Z'))
    
def main():
    overwrite = False
    for arg in sys.argv:
        if "-h" == arg or "--help" == arg:
            print_help()
        if "-f" == arg or "--force-overwrite" == arg:
            overwrite = True

    argv_size = len(sys.argv) if not overwrite else len(sys.argv) -1
    argv = sys.argv[1:argv_size]
    if len(argv) < 1 or len(argv) > 2:
        recommend_help()

    n = 0
    if len(argv) == 1:
        if not str(argv[0]).isdecimal() and not (argv[0][0] == '-' and str(argv[0][1:]).isdecimal()):
            recommend_help()
        n = int(argv[0])
    else:
        if len(argv[0]) != 1 or len(argv[1]) != 1:
            recommend_help()
        if not in_alphabet(argv[0]) or not in_alphabet(argv[1]):
            recommend_help()
        n = ord(str(argv[1]).lower()) - ord(str(argv[0]).lower())
    c_cypher(n, overwrite)

if __name__ == "__main__":
   main()

File: cypher.php
<?php

function print_help(){
    echo "Usage:\n";
    echo "a) php cypher.php YOUR_INT [OPTION] < YOUR_FILE_TO_ENCRYPT\n";
    echo "b) php cypher.php YOUR_SOURCE_CHAR YOUR_DESTINATION_CHAR [OPTION] < YOUR_FILE_TO_ENCRYPT\n";
    echo "[OPTION]:\n";
    echo "-h --help\n";
    echo "   prints usage (this).\n\n";
    echo "-f --force-overwrite\n";
    echo "   if destination file exists, overwrites it.\n";
    die();
}

function recommend_help(){
    echo "Bad arguments.\n";
    echo "Type: \"php cypher.php -h\" for help.\n";
    die();
}

function c_cypher($n, $overwrite){
    $n %= 26;
    if(file_exists("zasifrovana(". $n .").txt") && !$overwrite){
        echo "File already exists. If you want to overwrite, run the command with --force-overwrite or -f\n";
        die();
    }
    try{
        $out_file = fopen("zasifrovana(". $n .").txt", "w");
        if(! $out_file){
            throw new Exception();
        }
        $input = fopen('php://stdin', 'r');
        while($line = fgets($input)){
            $result = "";
            $chars = str_split($line);
            foreach($chars as $char){
                if (ord($char) >= ord('a') && ord($char) <= ord('z')){
                    $result .= chr((ord($char) + $n - ord('a')) % 26 + ord('a'));
                }
                else if(ord($char) >= ord('A') && ord($char) <= ord('Z')){
                    $result .= chr((ord($char) + $n - ord('A')) % 26 + ord('A'));
                }
                else{
                    $result .= $char;
                }
            }
            echo $result;
            fwrite($out_file, $result);
        }
    }
    catch(Exception $e){
        echo "Error has occured.\n";
        die();
    }
    finally{
        fclose($out_file);
    }
}
//echo count($argv);
if(count($argv) < 2 || count($argv) > 4){
    recommend_help();
}

$overwrite = false;
foreach($argv as $arg){
    //echo $arg;
    //echo "\n";
    if($arg == "-h" || $arg == "--help"){
        print_help();
    }
    if($arg == "-f" || $arg == "--force-overwrite"){
        $overwrite = true;
    }
}

$n = 0;
if(is_numeric($argv[1])){
    $n = intval($argv[1]);
}
else if(count($argv) > 2){
    if(strlen($argv[1]) != 1 || strlen($argv[2]) != 1){
        recommend_help();
    }
    if(!ctype_alpha($argv[1]) || !ctype_alpha($argv[2])){
        recommend_help();
    }
    $n = ord(strtolower($argv[2])) - ord(strtolower($argv[1]));
}
else{
    recommend_help();
}

c_cypher($n, $overwrite);

?>